packages/runtime/src/next/components/framework-provider/pages-router/HeadSnippet.tsx

```diff
@@ -19,7 +24,15 @@ export function HeadSnippet({ snippet }: { snippet: Snippet }) {
     }
   }, [snippet])
 
-  const headSnippetElement = snippetToElement(snippet)
+  const headSnippetElements = snippetToElements(snippet)
+  const [scripts, snippets] = partition(headSnippetElements, isScriptElement)
 
-  return <Head>{headSnippetElement}</Head>
+  return (
+    <>
+      <Head>{snippets}</Head>
+      {scripts.map(({ key, props }) => (
+        <Script key={key} {...props} strategy="beforeInteractive" />
+      ))}
+    </>
```

Comment:
I also considered and tried a more complex implementation below, thinking that it would preserve the relative order of elements within the snippet, but in practice it turned out to produce the exact same HTML, so I went with a simpler/cleaner implementation.

```tsx
  return (
    <>
      {headSnippetElements.map((element, i) =>
        isScriptElement(element) ? (
          <Script key={element.key} {...element.props} strategy="beforeInteractive" />
        ) : (
          <Head key={isElement(element) ? element.key : i}>{element}</Head>
        ),
      )}
    </>
  )
```

Reply:
Went back to the more complex but conceptually correct implementation for the sake of tests.